Framework/Kernel/inc/MantidKernel/case_insensitive_map.h

```diff
@@ -0,0 +1,31 @@
+#ifndef CASE_INSENSITIVE_MAP_H
```

Comment:
I think given that the alias class name is CamelCase then it would be better if the filename reflected that. Could you change it to `CaseInsensitiveMap`?